### PR TITLE
Create Authentication Page

### PR DESCRIPTION
**Pull request by smazouz42**
This PR addresses the task (https://linear.app/dark-orga/issue/DAR-16/add-authentication-fe) by adding login and sign-in pages using the `chadcn` components, React Hook Form, and Yup for validation.

**Summary:**
- Created login and sign-in pages.
- Integrated React Hook Form for form management.
- Used Yup for form validation.
- Styled pages using `chadcn` components.

### CODE REVIEW

File: src/components/LandingPage/index.tsx [hunk 1]
```diff
@@ -1,45 +1,39 @@
...
           type="video/mp4"
         />
         Your browser does not support the video tag.
       </video>
 
       <div className="flex flex-col text-center text-white gap-5 drop-shadow-lg">
-        <h1 className="~text-5xl/8xl font-bold">Welcome to Anitube</h1>
-        <div className="~text-lg/2xl mx-auto w-4/5 md:w-full">
+        <h1 className="text-5xl font-bold">Welcome to Anitube</h1>
+        <div className="text-lg mx-auto w-4/5 md:w-full">
```
Comment (4DeR99) [inline src/components/LandingPage/index.tsx]: why did you change this ?

File: src/components/LandingPage/index.tsx [hunk 2]
```diff
@@ -1,45 +1,39 @@
 'use client'
-import { Button } from '@nextui-org/react'
-import { useRouter } from 'next/navigation'
+import { useState } from 'react'
+
+import { LoginCard } from '../LoginCard'
 
 const LandingPage = () => {
-  const router = useRouter()
+  const [type, setType] = useState('login')
```
Comment (4DeR99) [inline src/components/LandingPage/index.tsx]: this should be inside the login/register component

File: src/components/LandingPage/index.tsx [hunk 3]
```diff
@@ -1,45 +1,39 @@
...
-          onClick={() => {
-            router.push('/login')
-          }}
-        >
-          Login
-        </Button>
+        <LoginCard
+          type={type}
+          setType={setType}
+        />
```
Comment (4DeR99) [inline src/components/LandingPage/index.tsx]: it doesn't handle only login so change its name to AuthCard

File: src/components/LoginCard/index.tsx
Comment (4DeR99) [inline src/components/LoginCard/index.tsx:1]: this should be split into 3 components one for login one for registration and one for password recovery
and name it AuthCard instead

Comment (smazouz42) [reply src/components/LoginCard/index.tsx:1]: fixed [b036224](https://github.com/dark-orga/anitube-website/pull/2/commits/b036224f01af39731e0a9ae8d060ce61d3b13dd9)

File: src/components/LandingPage/index.tsx (hunk 3 above)
Comment (smazouz42) [reply src/components/LandingPage/index.tsx]: fixed [b036224](https://github.com/dark-orga/anitube-website/pull/2/commits/b036224f01af39731e0a9ae8d060ce61d3b13dd9)

File: src/components/LandingPage/index.tsx (hunk 2 above)
Comment (smazouz42) [reply src/components/LandingPage/index.tsx]: fixed [b036224](https://github.com/dark-orga/anitube-website/pull/2/commits/b036224f01af39731e0a9ae8d060ce61d3b13dd9)

File: src/components/LandingPage/index.tsx (hunk 1 above)
Comment (smazouz42) [reply src/components/LandingPage/index.tsx]: fixed [b036224](https://github.com/dark-orga/anitube-website/pull/2/commits/b036224f01af39731e0a9ae8d060ce61d3b13dd9)

File: src/components/AuthDialog/index.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import Image from 'next/image'
+import * as React from 'react'
+import { useState } from 'react'
+
+import { Button } from '@/components/ui/button'
+import { Dialog, DialogContent, DialogTrigger } from '@/components/ui/dialog'
+
+import ForgotPassword from './ForgotPassword'
+import Login from './login'
+import SignUp from './SignUp/SignUp'
```
Comment (aanjaimi) [inline src/components/AuthDialog/index.tsx]: I think the file inside the SignUp directory should be named index

File: src/components/AuthDialog/SignUp/SignUp.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+import { signUpSchema } from '@/schemas/authSchemas'
+import { SignUpForm } from '@/types/auth'
+
+interface AuthDialogProps {
+  switchToLogin: () => void
+}
+
+const SignUp: React.FC<AuthDialogProps> = ({ switchToLogin }) => {
+  const signUpMethods = useForm<SignUpForm>({
+    resolver: yupResolver(signUpSchema as any),
```
Comment (aanjaimi) [inline src/components/AuthDialog/SignUp/SignUp.tsx]: You can remove the any here

**Review by aanjaimi — COMMENTED** [review]
Some simple changes need to be done

File: src/components/AuthDialog/ForgotPassword/index.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+const ForgotPassword: React.FC<AuthDialogProps> = ({ switchToLogin }) => {
+  const forgotPasswordMethods = useForm<ForgotPasswordForm>({
+    resolver: yupResolver(forgotPasswordSchema),
+  })
+
+  const onForgotPasswordSubmit: SubmitHandler<ForgotPasswordForm> = (data) => {
+    JSON.stringify(data)
+  }
```
Comment (4DeR99) [inline src/components/AuthDialog/ForgotPassword/index.tsx]: wrap this in a usecallback

File: src/components/AuthDialog/SignUp/SignUp.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+  })
+
+  const onSignUpSubmit: SubmitHandler<SignUpForm> = (data) => {
+    JSON.stringify(data)
+  }
```
Comment (4DeR99) [inline src/components/AuthDialog/SignUp/SignUp.tsx]: wrap this in a usecallback

File: src/components/AuthDialog/SignUp/SignUp.tsx
Comment (4DeR99) [inline src/components/AuthDialog/SignUp/SignUp.tsx:1]: change file name to index.tsx

File: src/components/AuthDialog/login/index.tsx [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+        >
+          Login
+        </Button>
+        <CardDescription className="text-center">
+          Don’t have an account?
+          <span
+            className="cursor-pointer text-purple-700 hover:text-purple-800"
+            onClick={switchToSignUp}
+          >
+            {' Register'}
```
Comment (4DeR99) [inline src/components/AuthDialog/login/index.tsx]: there is an unnecessary space

File: src/components/AuthDialog/index.tsx [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+                type === 'register'
+                  ? '/images/signIn.jpeg'
+                  : '/images/passwordReset.png'
+              }
+              alt="Illustration"
+              width={320}
+              height={320}
+            />
+          </div>
+        )}
```
Comment (4DeR99) [inline src/components/AuthDialog/index.tsx]: these images should be inside their card component

File: src/components/AuthDialog/index.tsx [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        {type === 'reset' && <ForgotPassword switchToLogin={switchToLogin} />}
+        {type === 'login' && (
+          <div className="hidden sm:flex sm:grow relative overflow-hidden items-center justify-center rounded-lg ">
+            <Image
+              src="/images/login.jpeg"
+              alt="Animated GIF"
+              layout="fill"
+            />
+          </div>
+        )}
```
Comment (4DeR99) [inline src/components/AuthDialog/index.tsx]: this one as well

File: tailwind.config.ts [hunk 11]
```diff
@@ -20,8 +20,17 @@ const config: Config = {
...
         foreground: 'var(--foreground)',
       },
+      animation: {
+        'move-up-down': 'moveUpDown 4s ease-in-out infinite',
+      },
+      keyframes: {
+        moveUpDown: {
+          '50%': { transform: 'translateY(-50px)' },
+        },
+      },
```
Comment (4DeR99) [inline tailwind.config.ts]: I don't see this used anywhere 
why is it here?

File: src/components/AuthDialog/SignUp/SignUp.tsx (hunk 5 above)
Comment (smazouz42) [reply src/components/AuthDialog/SignUp/SignUp.tsx]: The confirmPassword is optional in this validation schema. I just need to mark it as optional in the form to handle the error, instead of considering signUpSchema as any [74810f1](https://github.com/dark-orga/anitube-website/pull/2/commits/74810f1ca9ded7c30bb6f4f2e3fb8e9be2f46431)

File: src/components/AuthDialog/index.tsx (hunk 4 above)
Comment (smazouz42) [reply src/components/AuthDialog/index.tsx]: [74810f1](https://github.com/dark-orga/anitube-website/pull/2/commits/74810f1ca9ded7c30bb6f4f2e3fb8e9be2f46431)

File: tailwind.config.ts (hunk 11 above)
Comment (smazouz42) [reply tailwind.config.ts]: fixed [d4c46a4](https://github.com/dark-orga/anitube-website/pull/2/commits/d4c46a4fe8ffbf9e2f2a15e41c88e4b31ea311b6)

File: src/components/AuthDialog/index.tsx (hunk 10 above)
Comment (smazouz42) [reply src/components/AuthDialog/index.tsx]: fixed [d4c46a4](https://github.com/dark-orga/anitube-website/pull/2/commits/d4c46a4fe8ffbf9e2f2a15e41c88e4b31ea311b6)

File: src/components/AuthDialog/index.tsx (hunk 9 above)
Comment (smazouz42) [reply src/components/AuthDialog/index.tsx]: fixed [d4c46a4](https://github.com/dark-orga/anitube-website/pull/2/commits/d4c46a4fe8ffbf9e2f2a15e41c88e4b31ea311b6)

File: src/components/AuthDialog/SignUp/SignUp.tsx (hunk 7 above)
Comment (smazouz42) [reply src/components/AuthDialog/SignUp/SignUp.tsx]: fixed [d4c46a4](https://github.com/dark-orga/anitube-website/pull/2/commits/d4c46a4fe8ffbf9e2f2a15e41c88e4b31ea311b6)

File: src/components/AuthDialog/ForgotPassword/index.tsx (hunk 6 above)
Comment (smazouz42) [reply src/components/AuthDialog/ForgotPassword/index.tsx]: fixed [d4c46a4](https://github.com/dark-orga/anitube-website/pull/2/commits/d4c46a4fe8ffbf9e2f2a15e41c88e4b31ea311b6)

File: src/components/AuthDialog/SignUp/SignUp.tsx
Comment (smazouz42) [reply src/components/AuthDialog/SignUp/SignUp.tsx:1]: fixed [74810f1](https://github.com/dark-orga/anitube-website/pull/2/commits/74810f1ca9ded7c30bb6f4f2e3fb8e9be2f46431)

**Review by 4DeR99 — APPROVED** [review]
👍

**Comment by aanjaimi** [comment]
everything works great

**Review by aanjaimi — APPROVED** [review]
Everything works great
